src/main/java/SearchStockUseCase/ViewStockUseCaseInteractor.java

```diff
@@ -1,23 +1,77 @@
 package SearchStockUseCase;
-
 import APIInterface.StockAPIGateway;
 import APIInterface.StockAPIRequest;
 import APIInterface.StockAPIResponse;
+import yahoofinance.histquotes.HistoricalQuote;
+import yahoofinance.histquotes.Interval;
 
+import javax.swing.table.DefaultTableModel;
 import java.io.IOException;
+import java.text.DecimalFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.List;
+
+
+public class ViewStockUseCaseInteractor{
+    private StockAPIResponse stock;
+    private final String stockSymbol;
+    private Calendar from = Calendar.getInstance();
+    private final Interval stockPriceInterval = Interval.DAILY;
+
+    public ViewStockUseCaseInteractor(String symbol){
+        this.stockSymbol = symbol;
+    }
 
+    public void isValidStock() throws Exception {
+        //Checking is API throws error or not
+        try{
+            new StockAPIGateway().getPrice(new StockAPIRequest(this.stockSymbol));
+        }catch (IOException e){
+            throw new Exception("Invalid stock symbol");
+        }
+    }
 
```

Comment:
could serve to be useful adding Javadocs

Reply:
I have now added some documentation in the use case interactor